clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/converters/DeployAppengineAtomicOperationConverter.groovy

```diff
@@ -22,16 +22,43 @@ import com.netflix.spinnaker.clouddriver.appengine.deploy.ops.DeployAppengineAto
 import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
 import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperations
 import com.netflix.spinnaker.clouddriver.security.AbstractAtomicOperationsCredentialsSupport
+import com.netflix.spinnaker.clouddriver.appengine.deploy.exception.AppengineDescriptionConversionException
 import org.springframework.stereotype.Component
+import org.springframework.beans.factory.annotation.Autowired
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
 
 @AppengineOperation(AtomicOperations.CREATE_SERVER_GROUP)
 @Component
 class DeployAppengineAtomicOperationConverter extends AbstractAtomicOperationsCredentialsSupport {
+  @Autowired
+  ObjectMapper objectMapper = new ObjectMapper();
+
   AtomicOperation convertOperation(Map input) {
     new DeployAppengineAtomicOperation(convertDescription(input))
   }
 
   DeployAppengineDescription convertDescription(Map input) {
-    AppengineAtomicOperationConverterHelper.convertDescription(input, this, DeployAppengineDescription)
+    DeployAppengineDescription description = AppengineAtomicOperationConverterHelper.convertDescription(input, this, DeployAppengineDescription)
+
+    if (input.artifact) {
+      description.artifact = objectMapper.copy().convertValue(input.artifact, Artifact)
+      switch (description.artifact.type) {
+        case 'gcs/object':
+          description.repositoryUrl = description.artifact.reference;
+          break
+        case 'docker/image':
+          description.containerImageUrl = description.artifact.name;
+          break
+        default:
+          throw new AppengineDescriptionConversionException("Invalid artifact type for AppEngine deploy: ${description.artifact.type}")
```

Comment:
Very minor detail but I try to use `Appengine` or `App Engine` but not `AppEngine`.